coral-component-accordion/src/scripts/Accordion.js

```diff
@@ -329,6 +329,11 @@ class Accordion extends BaseComponent(HTMLElement) {
       }
     }
     
+    // set items level appropriately
+    if (item && item.level !== this.level) {
```

Comment:
Maybe you can use the attribute instead of the prop to be on the safe side as well.